packages/e2e-tests/helpers/utils.ts

```diff
@@ -92,31 +110,25 @@ export async function waitForRecordingToFinishIndexing(page: Page): Promise<void
   );
 
   const timelineCapsuleLocator = page.locator('[data-test-id="Timeline-Capsule"]');
-  try {
-    await waitFor(
-      async () => {
-        expect(await timelineCapsuleLocator.getAttribute("data-test-progress")).toBe("100");
-      },
-      {
-        retryInterval: 1_000,
-        timeout: 150_000,
-      }
-    );
-  } catch {
-    if (await page.locator('[data-test-id="SupportForm"]').isVisible()) {
-      const errorDetailsLocator = page.locator('[data-test-id="UnexpectedErrorDetails"]');
-      if (await errorDetailsLocator.isVisible()) {
-        throw new Error(`Recording crashed: ${await errorDetailsLocator.innerText()}`);
+
+  let supportFormErrorDetails: string | null = null;
+  await waitFor(
+    async () => {
+      supportFormErrorDetails = await getSupportFormErrorDetails(page);
+      if (supportFormErrorDetails) {
+        throw new UnrecoverableError(`Session failed: ${supportFormErrorDetails}`);
```

Comment:
This is an example of what we were talking about, @hbenl: Sometimes, things are unrecoverable, and we just don't want to retry.
But as per your suggestion, I added an explicit way to indicate unrecoverability, rather than assuming that most errors are unrecoverable.